File: scripts/copy-ui-components.py
#!/usr/bin/env python3
import shutil
import os
from pathlib import Path

src_ui = Path("/vercel/share/v0-project/src/components/ui")
dest_ui = Path("/vercel/share/v0-project/components/ui")

# Create destination if it doesn't exist
dest_ui.mkdir(parents=True, exist_ok=True)

# Copy all UI component files
if src_ui.exists():
    for file in src_ui.glob("*.tsx"):
        dest_file = dest_ui / file.name
        if not dest_file.exists():  # Only copy if it doesn't exist
            shutil.copy2(file, dest_file)
            print(f"Copied {file.name}")
        else:
            print(f"Skipped {file.name} (already exists)")
else:
    print(f"Source directory {src_ui} does not exist")

print("UI components migration complete!")

File: scripts/copy-ui-components.js
const fs = require('fs');
const path = require('path');

const srcDir = path.join(__dirname, '../src/components/ui');
const destDir = path.join(__dirname, '../components/ui');

// Create destination directory if it doesn't exist
if (!fs.existsSync(destDir)) {
  fs.mkdirSync(destDir, { recursive: true });
}

// Get all files in src/components/ui
const files = fs.readdirSync(srcDir);

let copiedCount = 0;
let skippedCount = 0;

files.forEach((file) => {
  const srcFile = path.join(srcDir, file);
  const destFile = path.join(destDir, file);

  // Skip if file already exists or is one we already wrote
  if (fs.existsSync(destFile)) {
    skippedCount++;
    return;
  }

  try {
    const content = fs.readFileSync(srcFile, 'utf-8');
    fs.writeFileSync(destFile, content);
    copiedCount++;
    console.log(`✓ Copied: ${file}`);
  } catch (error) {
    console.error(`✗ Error copying ${file}: ${error.message}`);
  }
});

console.log(`\nCopy complete! Copied: ${copiedCount}, Skipped: ${skippedCount}`);
